### PR TITLE
Make use of GitHub actions

### PR DESCRIPTION
**Pull request by ctron**
This PR allows to use GitHub actions in additions to CircleCI.

Everyone already has GitHub actions when using GitHub. So this makes it easier for people to contribute. As CircleCI forces you into creating an account and granting full access to all public repositories, just for viewing the test failures.

Signed-off-by: Jens Reimann <jreimann@redhat.com>

### CODE REVIEW

**Comment by floruschbaschan** [comment]
Thanks a lot for your contribution to Hawkbit (#998). We will look at it and then give feedback. But please keep in mind that currently many colleagues are having vacation so it might take it little bit longer as usual.

**Comment by schabdo** [comment]
Thanks a lot @ctron, highly appreciated! Indeed GitHub actions don't require anything and are well integrated. So for sure we could add them additionally to CircleCI and gain some experience. Depending of the outcome of [this discussion](https://discuss.circleci.com/t/authenticate-with-docker-to-avoid-impact-of-nov-1st-rate-limits) they may become our default CI very fast :)

**I do have only one remark:** Since `AmqpAuthenticationMessageHandlerIntegrationTest` tries to connect to a RabbitMQ broker (which is available on CircleCI) we will encounter errors in the output of Surefire. These errors are shown as "Check failure" annotations at the end of each action run, however build will succeed. FMPOV it would be cool to have this fixed before we introduce GitHub actions. From my own experience false-positives often lower the acceptance rate of something new :)

Long story short: I'll try to fix the mentioned test somehow but this will need more time. Meanwhile if somebody could lend a hand here this would awesome!

**Comment by schabdo** [comment]
I think I found a better way to get the mentioned issue addressed. Luckily there is a RabbitMQ action available which could be leveraged to execute the AMQP integration tests in a proper manner and consider the test results within a build (like we do on CircleCI). This should do the trick:

```
- uses: nijel/rabbitmq-action@v1.0.0
        with:
          rabbitmq version: '3.7-alpine'
```

@ctron having the RabbitMQ in place we're good to go for master. May I ask you to add this to this PR?

**Comment by diegorondini** [comment]
@ctron are you able to implement the requested changes?

**Comment by ctron** [comment]
> @ctron are you able to implement the requested changes?

Right now I am a bit busy with other stuff. However, feel free to amend the PR.

**Comment by avgustinmm** [comment]
outdated
